lang/typescript/src/utils/regions.ts

```diff
@@ -25,10 +54,34 @@ function isRegionYamlConfig(
   yamlConfig: ValidYamlType,
 ): yamlConfig is RegionYamlConfig {
   return (
-    yamlConfig !== null &&
-    typeof yamlConfig === 'object' &&
+    isYamlObject(yamlConfig) &&
     'code' in yamlConfig &&
-    typeof yamlConfig.code === 'string'
+    typeof yamlConfig.code === 'string' &&
+    (!('combined_address_format' in yamlConfig) ||
+      isCombinedAddressFormat(yamlConfig.combined_address_format as any))
+  );
+}
+
+// TODO: Figure out a cleaner way to do this or ditch it
+function isCombinedAddressFormat(
+  yamlProperty: Record<string, any>,
+): yamlProperty is CombinedAddressFormat {
+  return (
+    isYamlObject(yamlProperty) &&
+    Object.entries(yamlProperty).every(
+      ([key, value]) =>
+        (key === 'address1' &&
+          isYamlArray(value) &&
+          isYamlObject(value[0]) &&
+          'key' in value[0] &&
+          typeof value[0].key === 'string') ||
+        (key === 'address2' &&
+          isYamlArray(value) &&
+          isYamlObject(value[0]) &&
+          'key' in value[0] &&
+          typeof value[0].key === 'string') ||
+        key === 'zh-TW',
+    )
```

Comment:
Can ignore this, this was me _trying_ to strictly type this as a subset of valid yaml and I think this approach might be a fools errand unless I can find use something like [Zod](https://zod.dev/) for it to just write a schema for the region yaml.

The parser will error if it's invalid yaml that's probably Good Enough™️ anyway.